### PR TITLE
[DI] Create `TechDocs` search result item extension

### PR DESCRIPTION
**Pull request by camilaibs**
## Hey, I just made a Pull Request!

Migrate the `TechDocsSearchResultItemExtension` to the declarative integration format.

### Known limitations:

- It doesn't support icon customization for now.

### Usage example:

> [!Note]
> Currently, it is in an experimental stage and is not recommended for production

For frontend applications that support declarative integration, it is only necessary to install the following packages:

```sh
# install the catalog, search, and techdocs plugins packages
yarn add @backstage/plugin-catalog @backstage/plugin-search @backstage/plugin-techdocs
```

Optionally you can customize the search results via `app-config.yaml` file:

```yaml
# app-config.yaml
app:
  experimental:
    packages: 'all' # ✨

  extensions:
    - plugin.search.result.item.techdocs:
        config:
          noTrack: true
          lineClamp: 3
          asListItem: false
          asLink: false
          title: 'Custom title'
```
### Testing

Install the branch dependencies and start both frontend and backend declarative applications:

```sh
# Installing branch dependencies
yarn install

# Starting the backend application
cd packages/backend-next
yarn start

# Starting the frontend application
cd packages/app-next
yarn start
```
After that, navigate to http://locahost:3000 and click on the `Search` sidebar item:

image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/techdocs/src/alpha.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  createPlugin,
+  createSchemaFromZod,
+} from '@backstage/frontend-plugin-api';
+import { createSearchResultListItemExtension } from '@backstage/plugin-search-react/alpha';
+
+/** @alpha */
+export const TechDocsSearchResultListItemExtension =
+  createSearchResultListItemExtension({
+    id: 'techdocs',
+    at: 'plugin.search.page/items',
```
Comment (Rugvip) [inline plugins/techdocs/src/alpha.tsx:28]: I just realized, this should probably be the default

**Review by Rugvip — APPROVED** [review]
👍

File: plugins/techdocs/src/alpha.tsx (hunk 1 above)
Comment (camilaibs) [reply plugins/techdocs/src/alpha.tsx:28]: Oh that's a good idea, I'll do that 🙌🏻

Comment (camilaibs) [reply plugins/techdocs/src/alpha.tsx:28]: Added here 🙏🏻 
https://github.com/backstage/backstage/pull/20265

**Review by agentbellnorm — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**